server/app.ts

```diff
@@ -128,6 +129,30 @@ app.get("/api/history", (req, res) => {
         .then(j => res.json(j))
 })
 
+export const onSendEmail: (cfg: gitfs.Config) => express.RequestHandler = cfg => 
+    async (req, res) => {
+    const msg = req.body as Message
+    const hostHeaderIndex = req.rawHeaders.indexOf('Host') + 1;
```

Comment:
You need to check Origin header. Host will always match. You could also try Referer. 

Also i think you want to throttle the request to say 1 per minute from a given IP.